### PR TITLE
Include attribute values in target search text filter

### PR DESCRIPTION
**Pull request by stschake**
Most of our devices are registered through plug&play so the majority of useful information is in the controller attributes. This PR also applies the search text filter to the attribute values.

### CODE REVIEW

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java [hunk 1]
```diff
@@ -148,6 +150,35 @@ private TargetSpecifications() {
...
+     * @param searchText
+     *            to be filtered on
+     * @return the {@link Target} {@link Specification}
+     */
+    public static Specification<JpaTarget> likeAttributeValue(final String searchText) {
+        return (targetRoot, query, cb) -> {
+            final String searchTextToLower = searchText.toLowerCase();
+            final MapJoin<JpaTarget, String, String> attributeMap = targetRoot.join(JpaTarget_.controllerAttributes,
+                    JoinType.LEFT);
+            return cb.or(cb.like(cb.lower(attributeMap.value()), searchTextToLower));
```
Comment (stefbehl) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java]: In order to eliminate duplicates in the search result, you need to set the distinct flag to true:
`query.distinct(true);`

You can easily recreate this by creating a target with a controller ID that is equal to one of the attribute values. In that case the target would appear twice in the list.

**Comment by schabdo** [comment]
@hawkbit-bot verify please

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java (hunk 1 above)
Comment (stschake) [reply hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java]: I've updated the test to check for that, but can't seem to reproduce the issue with it.

Comment (stefbehl) [reply hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java]: Thanks for updating the test! I could easily recreate this in the UI a few weeks ago but I will double-check...

Comment (stefbehl) [reply hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java]: Hi Stefan,
The new test fails for me! However, if I add query.distinct(true) as proposed above, everything works fine :-) Would be great if you could add that.
Thanks and regards.

Comment (stschake) [reply hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java]: It seems to pass on CircleCI? Happy to add it, just wondering what's going on there.

Comment (stefbehl) [reply hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java]: That's indeed weird. I ran the tests in my Eclipse and there it failed. After adding the distinct flag that tests passed... Anyway, thanks for adding this line!

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java [hunk 2]
```diff
@@ -148,6 +150,36 @@ private TargetSpecifications() {
...
+     *            to be filtered on
+     * @return the {@link Target} {@link Specification}
+     */
+    public static Specification<JpaTarget> likeAttributeValue(final String searchText) {
+        return (targetRoot, query, cb) -> {
+            final String searchTextToLower = searchText.toLowerCase();
+            final MapJoin<JpaTarget, String, String> attributeMap = targetRoot.join(JpaTarget_.controllerAttributes,
+                    JoinType.LEFT);
+            query.distinct(true);
+            return cb.or(cb.like(cb.lower(attributeMap.value()), searchTextToLower));
```
Comment (stefbehl) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java]: The "cb.or" at the beginning of the return statement is not needed. Please remove it. Thanks

Comment (stschake) [reply hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java]: Done. I've also removed the whitespace changes.

**Pull request merged**